### PR TITLE
[ROCm] Update ROCm CI to python3.6

### PR DESCRIPTION
**Pull request by iotamudelta**
Given that python 2.7 will be EOL'd on Jan 1, 2020 and we have models depending on python3.5+, we'd like to update the ROCm CI across the board to python3.6.

This PR adds the skip tests and some semantic changes for PyTorch.

Open tasks/questions:
* RoiAlignTest.CheckCPUGPUEqual fails in the Caffe2 unit tests. Is this something expects / can be skipped?
* for testing, I've used update-alternatives on CentOS/Ubuntu to select python == python 3.6. Is this the preferred way?

### CODE REVIEW

**Comment by ezyang** [comment]
> RoiAlignTest.CheckCPUGPUEqual fails in the Caffe2 unit tests. Is this something expects / can be skipped?

Long standing flaky test #20488 Please feel free to disable it.

> for testing, I've used update-alternatives on CentOS/Ubuntu to select python == python 3.6. Is this the preferred way?

That's fine, and what we do in our docker images.

File: test/test_nn.py [hunk 1]
```diff
@@ -5267,7 +5267,7 @@ def test_KLDivLoss_batch_mean(self):
 
         self.assertEqual(l, expected)
 
-    @unittest.skipIf(not (TEST_CUDNN and TEST_CUDNN_VERSION >= 7000), "needs cudnn >= 7.0")
+    @unittest.skipIf(not (TEST_CUDNN and (TEST_CUDNN_VERSION if TEST_CUDNN_VERSION else 0) >= 7000), "needs cudnn >= 7.0")
```
Comment (ezyang) [inline test/test_nn.py]: This is kind of goofy can we just unconditionally set `TEST_CUDNN_VERSION` to zero if we don't have it set?

**Review by ezyang — APPROVED** [review]
This seems fine.

File: test/test_cuda.py [hunk 2]
```diff
@@ -2339,6 +2339,7 @@ def plan_cache_max_size(n, device=None):
                             self.assertEqual(torch.backends.cuda.cufft_plan_cache.max_size, 10)  # default is cuda:0
                         self.assertEqual(torch.backends.cuda.cufft_plan_cache.max_size, 11)  # default is cuda:1
 
+    @skipIfRocm
```
Comment (bddppq) [inline test/test_cuda.py]: Do you know why do these tests fail?

**Comment by bddppq** [comment]
> update-alternatives on CentOS/Ubuntu

In Caffe2 docker images (both PyTorch and Caffe2 ROCm CI are using Caffe2 docker images), we don't override the default python, and instead require explicit uses of `python3`/`python3.6` etc.

**Comment by iotamudelta** [comment]
@bddppq thanks, I'll then enable it that way for both of them.

File: test/test_cuda.py (hunk 2 above)
Comment (iotamudelta) [reply test/test_cuda.py]: not entirely clear to me. I suspect that the use of librosa in stft may be an issue here.

Comment (bddppq) [reply test/test_cuda.py]: I see. In this case could you add a comment here to indicate this test passes in py2 but fails in py3?

**Review by bddppq — APPROVED** [review]
LGTM

**Comment by ezyang** [comment]
@bddppq are you going to handle landing this?

**Comment by iotamudelta** [comment]
How are we updating the CI infrastructure itself (i.e., the docker containers) to switch it over?

**Comment by yf225** [comment]
Note that this is breaking Windows CI:

20:52:19 +++ which python3
20:52:19 which: no python3 in (/mingw64/bin:/usr/bin:/c/Windows/system32/config/systemprofile/bin:/c/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v9.2/bin:/c/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v9.2/libnvvp:/c/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v9.0/bin:/c/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v9.0/libnvvp:/c/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v8.0/bin:/c/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v8.0/libnvvp:/c/Python37/Scripts:/c/Python37:/c/Python27:/c/Python27/Scripts:/c/Windows/system32:/c/Windows:/c/Windows/System32/Wbem:/c/Windows/System32/WindowsPowerShell/v1.0:/c/Program Files/Amazon/cfn-bootstrap:/c/ProgramData/chocolatey/bin:/c/Program Files/OpenSSH-Win64:/bin:/c/Program Files/Java/jdk1.8.0_162/bin:/c/Program Files/Amazon/AWSCLI/bin:/c/Program Files/Amazon/AWSCLI)

**Comment by bddppq** [comment]
@yf225 Thanks, reverting

**Comment by iotamudelta** [comment]
@yf225 @bddppq do I understand this correctly that the Windows CI is named `py3` but does not provide a `python3` executable? Is `python` aliased to be python3?

**Pull request closed without merging**